sympy/algebras/quaternion.py

```diff
@@ -459,9 +459,9 @@ def exp(self):
         q = self
         vector_norm = sqrt(q.b**2 + q.c**2 + q.d**2)
         a = exp(q.a) * cos(vector_norm)
-        b = exp(q.a) * sin(vector_norm) * q.b / vector_norm
-        c = exp(q.a) * sin(vector_norm) * q.c / vector_norm
-        d = exp(q.a) * sin(vector_norm) * q.d / vector_norm
+        b = exp(q.a) * sin(vector_norm) * q.b / (Piecewise((vector_norm, vector_norm > 0), (1, True)))
+        c = exp(q.a) * sin(vector_norm) * q.c / (Piecewise((vector_norm, vector_norm > 0), (1, True)))
+        d = exp(q.a) * sin(vector_norm) * q.d / (Piecewise((vector_norm, vector_norm > 0), (1, True)))
```

Comment:
There should be tests checking that these Piecewise are returned.

Rather than `vector_norm > 0` I think it would be better to use `Ne(vector_norm, 0)`.

The common factor could be precomputed like:
```python
scale = sin(vector_norm) / Piecewise((vector_norm, Ne(vector_norm, 0)), (1, True))
```
Actually this Piecewise is just `sinc` so I think it would be better to use `sinc` here.

Reply:
I've made the changes , any more changes to make?